biochemist-python/chapters/Processing_Multiple_Files_and_Writing_Files.ipynb

```diff
@@ -0,0 +1,427 @@
+    "    \n",
+    "The goal of this exercise is to extract the Enzyme Commission Class for a series of enzyme structures in PDB files and write them to a new text file called 'EC_class.txt'. The files are located in the `data/enzymes` folder. If you open any of these files and search for the term \"EC:\" you will find a listing that looks like this: \n",
+    "    \n",
+    "    COMPND   6 EC: 1.2.1.13;\n",
+    "    \n",
+    "You are probably familiar with these numbers, but just in case - the Enzyme Commission class tells you the function of an enzyme in a hierarchical format. You can learn more at the [BRENDA EC Explorer](https://www.brenda-enzymes.org/ecexplorer.php?browser=1&f[nodes]=21&f[action]=close&f[change]=21#21).  \n",
+    "  \n",
+    "**Your assignment** is to parse the files in the `data/enzymes` folder and write a new file named `EC_class.txt` that contains the PDB ID and EC class for each of these enzymes. When you open the file in your text editor, it should look like this:\n",
+    "\n",
+    "7tim \t  5.3.1.1  \n",
+    "6zt7 \t  3.2.1.55  \n",
+    "5eu9 \t  4.2.1.11  \n",
+    "3iva \t  2.1.1.13  \n",
+    "2pkr \t  1.2.1.13  \n",
+    "3vnd \t  4.2.1.20  \n",
+    "5veu \t  1.14.14.1\n",
+    "    \n",
+    "```{admonition} Hint\n",
+    ":class: dropdown\n",
+    "It helps when you are writing code to break up what you have to do into steps. Overall, we want to get information from the file. How do we do that?\n",
+    " \n",
+    "If you think about the steps you will need to do this assignment you might come up with a list that is like this: \n",
+    "\n",
+    "1. Open the file for reading.\n",
+    "1. Read the data in the file.\n",
+    "1. Loop through the lines in the file.\n",
+    "    1. Read the files to gain access to the information we want.\n",
+    "    1. Extract the desired information and write it to a file.\n",
+    "\n",
+    "It can be helpful when you code to write out these steps and work on it in pieces. Try to write the code using these steps. Note that as you write the code, you may come up with other steps!  \n",
+    "  \n",
+    "First, think about what you have to do for step 1, and write the code for that. Next, think about how you would do step 2 and write the code for that. You can troubleshoot each step using print statments. \n",
+    "  \n",
+    "The steps build on each other, so you can work on getting each piece written before moving on to the next.  \n",
+    "```\n",
+    "\n",
+    "```{admonition} Solution\n",
+    ":class: dropdown\n",
+    "\n",
+    "```python\n",
+    "# *Open the file for reading*\n",
+    "file_location = os.path.join('data', 'enzymes', '*.pdb')\n",
+    "\n",
+    "# Make sure you can see all of the files*\n",
+    "filenames = glob.glob(file_location)\n",
+    "# print(filenames)  #Once you're sure it's working, comment this line out.\n",
+    "\n",
+    "# *Loop through the lines in the files*\n",
```

Comment:
Nice exercise/project. I like that it uses multiple files (the subject of the lesson). I also didn't know about enzyme commission classes.

Note that if we switch to opening files with `with`, this solution should be adjusted.

Reply:
Just switched it over to using _with open_